### PR TITLE
Fixed remove bag()

### PR DESCRIPTION
**Pull request by we-ai**
The removeBag() function was updated to handle new box structure.

### CODE REVIEW

File: utils/firestore.js [hunk 1]
```diff
@@ -764,33 +763,43 @@ const updateBox = async (id, data, loginSite) => {
...
-        let box = snapshot.docs[0];
-        let data = box.data()
-        let currBags = data.bags;
-        let bagIds = Object.keys(currBags);
-        for(let i = 0; i < bags.length; i++){
-            if(currBags.hasOwnProperty(bags[i])){
-                delete currBags[bags[i]]
+        let doc = snapshot.docs[0];
+        let box = doc.data();
+        let bagConceptIDs = ["147157381", "147157382", "147157383", "147157384", "147157385", "147157386", "147157387", "147157388", "147157389", "147157390", "147157391", "147157392", "147157393", "147157394", "147157395"];
```
Comment (anthonypetersen) [inline utils/firestore.js:774]: Thoughts on putting this data structure into a shared function similar to what you did with the site codes?

File: utils/firestore.js [hunk 2]
```diff
@@ -764,33 +763,43 @@ const updateBox = async (id, data, loginSite) => {
...
             }
-            else{
-                console.log(bags[i] + ' NOT FOUND')
+        }
+
+        let bagConceptIDIndex = 0;
+        for (let k of Object.keys(box)) { 
+            if (bagConceptIDs.includes(k)) {
+                const currBagConceptID = bagConceptIDs[bagConceptIDIndex];
+                if (currBagConceptID === k) continue;
```
[the remark this answers is not in the document]
Comment (we-ai) [reply utils/firestore.js:790]: The loop is to re-organize bag concept IDs in box after deleting a bag. For example, if a box has  "147157381", "147157382", "147157383", when "147157381" is deleted, the rest 2 bags will be re-named as "147157381" and "147157382".

File: utils/firestore.js (hunk 1 above)
Comment (we-ai) [reply utils/firestore.js:774]: We can move it to shared.js file.

Comment (anthonypetersen) [reply utils/firestore.js:774]: Perfect, thanks.

**Review by Gbarra9 — APPROVED** [review]

**Pull request merged**